sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/PushNotificationClient.java

```diff
@@ -185,4 +214,77 @@ private ChatEvent parsePushNotificationEvent(ChatEventType chatEventType, ChatPu
         this.logger.info(" Result: " + jsonString);
         return NotificationUtils.toEventPayload(chatEventType, jsonString);
     }
+
+    private void scheduleRegistrationRenew(final long delayMs, final int tryCount) {
+        if (!this.isPushNotificationsStarted) {
+            this.logger.info("Push notification has already been stopped! No need to renew registration.");
+            return;
+        }
+
+        if (this.deviceRegistrationToken == null) {
+            stopPushNotifications();
+            Throwable throwable = new Throwable("No device registration token stored!");
```

Comment:
Should we make this an `IllegalStateException` or something a bit more specific? What do you think @srnagar, @JonathanGiles?
```suggestion
            Throwable throwable = new IllegalStateException("No device registration token stored!");
```